src/Cake.Common/Build/TFBuild/Data/TFBuildPublishTestResultsData.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Build.TFBuild.Data
+{
+    /// <summary>
+    ///  Provides TF Build agent info for publishing test results
+    /// </summary>
+    public class TFBuildPublishTestResultsData
+    {
+        /// <summary>
+        /// Gets or Sets the type test runner the results are formated in
+        /// </summary>
+        public TFTestRunnerType? TestRunner { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of Test Result files to publish.
+        /// </summary>
+        public string[] TestResultsFiles { get; set; }
```

Comment:
Any thoughts about making this `IReadOnlyList<string>`? I'm doing `.ToArray()` in my Cake script but I wouldn't have to.